miner/worker.go

```diff
@@ -613,6 +619,31 @@ func (w *worker) resultLoop() {
 	}
 }
 
+// timestampLoop is a loop that updates the timestamp used for blocks when it
+// is stale by more than a certain threshold.
+// TODO: Re-think this as everything comes together more.
+func (w *worker) timestampLoop() {
+	timer := time.NewTimer(0)
+
+	for {
+		select {
+		case <-timer.C:
+			currentTime := time.Now().Unix()
+			skew := currentTime - w.chain.CurrentTimestamp()
+			if skew > maxClockSkewSeconds {
+				newTime := currentTime - timestampDelaySeconds
+				w.chain.SetCurrentTimestamp(newTime)
+				timer.Reset((maxClockSkewSeconds - timestampDelaySeconds) * time.Second)
+				log.Debug(fmt.Sprintf("Timestamp was above max clock skew of %d seconds, so updating to current time - %d seconds (%d)", maxClockSkewSeconds, timestampDelaySeconds, newTime))
```

Comment:
Not super opinionated on this but the style of the codebase usually uses the key/value contexts as additional arguments to the `log` statement like:

```go
log.Debug("timestamp above max clock skew", "maxSkew", maxClockSkewSeconds, "overBy", timestampDelaySeconds, "newTime", newTime)

```

This would result in something that looks like:

```
DEBUG[06-17|21:58:10] timestamp above max clock skew                       maxSkew=10 overBy=5 newTime=135
```

One benefit would be the ability to not need `fmt` in this file. What do you think?
See https://github.com/ethereum/go-ethereum/tree/master/log